superset/db_engine_specs/ocient.py

```diff
@@ -237,10 +237,12 @@ def fetch_data(cls, cursor, lim=None):
 
         if columns_to_sanitize:
             # At least 1 column has to be sanitized.
-            for row in rows:
-                for info in columns_to_sanitize:
-                    # Modify the element in-place.
-                    v = row[info.column_index]
-                    row[info.column_index] = info.sanitize_func(v)
-
+            def do_nothing(x): 
+                return x
+            
+            sanitization_functions = [do_nothing for _ in range(len(rows[0]))]
```

Comment:
is the row list also immutable?

Reply:
```python
funcs = to_dict(columns_to_sanitize)

for i, row in enumerate(rows):
  rows[i] = [info.sanitize_func(val) for j, val in enumerate(row) if j in funcs else val]
```
Another alternative, though I'm not sure it's any more readable...

Reply:
No it is not. It is simply a list